ext/hal/libmetal/libmetal/lib/io.c

```diff
@@ -33,7 +33,7 @@ void metal_io_init(struct metal_io_region *io, void *virt,
 int metal_io_block_read(struct metal_io_region *io, unsigned long offset,
 	       void *restrict dst, int len)
 {
-	void *ptr = metal_io_virt(io, offset);
+	unsigned char *ptr = (unsigned char *)metal_io_virt(io, offset);
```

Comment:
@mped-oticon, this makes changes to ext/hal/ , and that's usually a good reason to make that a separate commit, and review would usually include a question "did you submit these changes upstream"?

@galak: FYI

Reply:
@pfalcon See mine and Carles' comment above.
Should I update our copy of libmetal (where this has been fixed -- in a slightly different way than this PR)?

Reply:
> Should I update our copy of libmetal (where this has been fixed -- in a slightly different way than this PR)?

Sure, that's the best solution, please coordinate with @galak who maintains Zephyr's copy. Also cc: @wjliang, upstream maintainer.